agent/__init__.py

```diff
@@ -209,28 +209,102 @@ def get_ca_cert(debug=False):
+        requester_name, req_type, response.status_code))
+    print("wott-agent :: mtls_post_request :: [RECEIVED] {} {}: {}".format(
+        requester_name, req_type, response.content))
+
+
+def mtls_post_request(req, json=None, debug=False, dev=False, requester_name=None, debug_on_ok=False):
+    """
+    MTLS POST Request wrapper function.
+    :param req: request url string (without endpoint)
+    :param json: json data, None by default
+    :param debug: if true then log error messages
+    :param dev: if true use dev endpoint and dev headers
+    :param requester_name: displayed requester id for error messages
+    :param debug_on_ok: if true with debug then log successful response status/content too
+    :return: response or None (if there was exception raised)
+    """
+    try:
+        r = requests.post(
+            '{}{}'.format(MTLS_ENDPOINT, req),
+            cert=(CLIENT_CERT_PATH, CLIENT_KEY_PATH),
+            json=json,
+            headers=get_mtls_header(debug=debug, dev=dev)
+        )
+
+        if (debug_on_ok or not r.ok) and debug:
+            mtls_req_error_log(req, 'POST', requester_name, r)
+        return r
+
+    except Exception as e:
+        print('wott-agent :: mtls_post_request :: rises exception: {}'.format(e))
+        return None
+
+
+def mtls_get_request(req, debug=False, dev=False, requester_name=None, debug_on_ok=False):
+    """
+    MTLS GET Request wrapper function.
+    :param req: request url string (without endpoint)
+    :param debug: if true then log error messages
+    :param dev: if true use dev endpoint and dev headers
+    :param requester_name: displayed requester id for error messages
+    :param debug_on_ok: if true with debug then log successful response status/content too
+    :return: response or None (if there was exception raised)
+    """
+    try:
+        r = requests.get(
+            '{}{}'.format(MTLS_ENDPOINT, req),
+            cert=(CLIENT_CERT_PATH, CLIENT_KEY_PATH),
+            headers=get_mtls_header(debug=debug, dev=dev)
+        )
+
+        if (debug_on_ok or not r.ok) and debug:
+            mtls_req_error_log(req, 'GET', requester_name, r)
+        return r
+
+    except Exception as e:
+        print('wott-agent :: mtls_get_request :: rises exception: {}'.format(e))
+        return None
+
+
 def get_claim_token(debug=False, dev=False):
     setup_endpoints(dev, debug)
     can_read_cert()
-    try:
-        response = requests.get('{}/v0.2/claimed'.format(MTLS_ENDPOINT), cert=(CLIENT_CERT_PATH, CLIENT_KEY_PATH),
-                                headers={'SSL-CLIENT-SUBJECT-DN': 'CN=' + get_device_id(),
-                                         'SSL-CLIENT-VERIFY': 'SUCCESS'} if dev else {})
-    except requests.exceptions.ConnectionError:
+
+    response = mtls_get_request('/v0.2/claimed', debug=debug, dev=dev,
+                                requester_name="Get Device Claim Info")
+    if not response or not response.ok:
```

Comment:
not `if not response`, but `if not response is None`

Reply:
fixed